app/models/split.rb

```diff
@@ -62,15 +64,15 @@ def reweight!(weighting_registry)
   end
 
   def assignment_count_for_variant(variant)
-    assignments.where(variant: variant).count(:id)
+    Assignment.from(assignments.for_presentation, :assignments).where(variant: variant).count(:id)
   end
 
   def build_decision(params = {})
-    Decision.new({ split: self }.merge(params))
+    Decision.new(params.merge(split: self))
   end
 
-  def create_decision!(params = {})
-    build_decision(params).tap(&:save!)
+  def create_decision!(params)
+    build_decision(params).save!
```

Comment:
```suggestion
    build_decision(params).tap(&:save!)
```
This shouldn't have been modified.